crates/sui-bridge-indexer/src/eth_worker.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::postgres_manager::{write, PgPool};
+use crate::{BridgeDataSource, TokenTransfer, TokenTransferData, TokenTransferStatus};
+use ethers::providers::Provider;
+use ethers::providers::{Http, Middleware};
+use ethers::types::Address as EthAddress;
+use std::sync::Arc;
+use sui_bridge::abi::{EthBridgeEvent, EthSuiBridgeEvents};
+use sui_bridge::types::EthLog;
+
+pub async fn process_finalized_eth_events(
+    mut eth_events_rx: mysten_metrics::metered_channel::Receiver<(EthAddress, u64, Vec<EthLog>)>,
+    provider: Arc<Provider<Http>>,
+    pool: &PgPool,
+) {
+    while let Some(event) = eth_events_rx.recv().await {
+        for log in event.2.iter() {
+            let eth_bridge_event = EthBridgeEvent::try_from_eth_log(log);
+            if eth_bridge_event.is_none() {
+                continue;
+            }
+            let bridge_event = eth_bridge_event.unwrap();
+            let block_number = log.block_number;
+            let block = provider.get_block(log.block_number).await.unwrap().unwrap();
+            let timestamp = block.timestamp.as_u64() * 1000;
+            let transaction = provider
+                .get_transaction(log.tx_hash)
+                .await
+                .unwrap()
+                .unwrap();
+            let gas = transaction.gas;
+            let tx_hash = log.tx_hash;
+
+            println!("Observed Finalized Eth bridge event: {:#?}", bridge_event);
```

Comment:
in general we shouldn't use `print` or `println` in production code unless in cli.
use `debug!`, `info` etc instead